drivers/impedance-analyzer/ad5933/ad5933.c

```diff
@@ -457,3 +457,31 @@ double ad5933_calculate_impedance(struct ad5933_dev *dev,
 
 	return impedance;
 }
+
+/***************************************************************************//**
+ * @brief Selects the number of settling cycles of the device.
+ *
+ * @param dev		 - The device structure.
+ * @param number_cycles	 - 9-bit number of cycles to wait before triggering ADC
+ *
+ * @param multiplier - Multiply number of cycles by X1, X2 or X4
+ *                Example: AD5933_SETTLING_X1
+ *                         AD5933_SETTLING_X2
+ *                         AD5933_SETTLING_X4
+ *
+ * @return None.
+*******************************************************************************/
+void ad5933_set_settling_time(struct ad5933_dev *dev,
+			      uint8_t multiplier,
+			      uint16_t number_cycles)
+{
+	if ((multiplier != AD5933_SETTLING_X2) && (multiplier != AD5933_SETTLING_X4))
+		multiplier = AD5933_SETTLING_X1;
+
+	ad5933_set_register_value(dev,
+				  AD5933_REG_SETTLING_CYCLES,
+				  number_cycles | (multiplier << 9),
+				  2);
+	/* Store the last settings made. */
+	dev->current_settling = number_cycles;
```

Comment:
As I see in your changes, current_settiling is not used anywhere. Can you remove it?

Reply:
Shall I remove this as part of new commit or force push the update ?

Reply:
Forcee push,  and also from ad5933_dev and ad5933_init_param

Reply:
Done.